### PR TITLE
Bug 1889573: Fix nil map in updating kibana resource

### PR DESCRIPTION
**Pull request by huikang**
- add test cases

/cc @jcantrill 
/assign @ewolinetz 

### Links
- Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1889573
- JIRA: https://issues.redhat.com/browse/OCPBUGSM-19482?filter=-1

### CODE REVIEW

File: pkg/utils/utils_test.go [hunk 1]
```diff
@@ -26,3 +29,31 @@ func TestLookupEnvWithDefaultUndefined(t *testing.T) {
...
+	} {
+		{
+			&v1.ResourceRequirements{},
+			&v1.ResourceRequirements{},
+			false,
+		},
+		{
+			&v1.ResourceRequirements{},
+			&v1.ResourceRequirements{
+				Limits: map[v1.ResourceName]resource.Quantity{"cpu": *resource.NewMilliQuantity(1000, resource.DecimalSI)},
```
Comment (ewolinetz) [inline pkg/utils/utils_test.go:47]: can we also add a test case with `Requests` ?

File: pkg/utils/utils_test.go [hunk 2]
```diff
@@ -26,3 +29,31 @@ func TestLookupEnvWithDefaultUndefined(t *testing.T) {
...
+			},
+			true,
+		},
```
Comment (ewolinetz) [inline pkg/utils/utils_test.go:50]: can we also add tests for the inverse of this? so if desired is empty

**Comment by ewolinetz** [comment]
:+1: looks good, just looking for additional unit tests to cover more scenarios

File: pkg/utils/utils_test.go (hunk 1 above)
Comment (huikang) [reply pkg/utils/utils_test.go:56]: Done

File: pkg/utils/utils_test.go (hunk 2 above)
Comment (huikang) [reply pkg/utils/utils_test.go:59]: Done

**Review by huikang — COMMENTED** [review]
Thanks for the feedback.
